internal/proto/local/plugin/host.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package plugin.sdk.proto;
+
+option go_package = "github.com/hashicorp/boundary/internal/plugin/sdk/proto;proto";
+
+import "google/protobuf/struct.proto";
+
+// Host contains all fields related to a host returned from a plugin.
+message Host {
+  // The primary address for the host result.
+  string address = 10;
+
+	// Name of the host.
+	string name = 20;
+
+	// The host catalog of which this host is a part.
+	string host_catalog_id = 30;
+
+	// The list of host sets containing this host.
+	repeated string host_set_ids = 40;
+
+  // The attributes that are applicable to the specific host. This
+  // may or may not be opaque and can be used for further
+  // post-processing by Boundary.
+	google.protobuf.Struct attributes = 50;
```

Comment:
Please label this as optional in the comment above.

This seems useful for allowing the plugin to give options around filtering at list time and maybe when building the filters that go into the host set, right?  Is there any other programmatic purpose you see for this field?

Reply:
Yeah, this would be the "for future use" metadata to be used for filtering. Figured it would be useful to scaffold this now, but maybe not?

Reply:
Going to keep it as available. Fixed the comments though, and made them more descriptive to the purpose. I think we could possibly surface this in API call results, which would be good just for information purposes at least!